### PR TITLE
Feature/gh 5477

### LINKED ISSUE #5477: Add "order return request" details page

### PR DESCRIPTION
**Pull request by WeizhengSap**
close #5477

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/myaccount/order/cancellations-returns/cancel-or-return.service.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ import {
   GlobalMessageService,
   GlobalMessageType,
   UserOrderService,
+  OrderReturnRequestService,
```
Comment (plabadie) [inline projects/storefrontlib/src/cms-components/myaccount/order/cancellations-returns/cancel-or-return.service.ts]: `cancel-or-return.service.ts ` could be in a `services` folder.  Your call.

File: projects/storefrontlib/src/cms-components/myaccount/order/cancellations-returns/cancel-or-return.service.ts [hunk 2]
```diff
@@ -41,11 +42,35 @@ export class OrderCancelOrReturnService {
...
+            cxRoute: 'returnRequestDetails',
+            params: { rma: state.value.rma },
+          });
+        }
+      }),
+      map(state => state.loading),
+      share()
+    );
+  }
+
```
[the remark this answers is not in the document]
Comment (WeizhengSap) [reply projects/storefrontlib/src/cms-components/myaccount/order/cancellations-returns/cancel-or-return.service.ts]: refactored

File: projects/storefrontlib/src/cms-components/myaccount/order/cancellations-returns/cancel-or-return.service.ts [hunk 3]
```diff
@@ -41,11 +42,35 @@ export class OrderCancelOrReturnService {
     );
   }
 
+  get isReturning$(): Observable<boolean> {
```
[the remark this answers is not in the document]
Comment (WeizhengSap) [reply projects/storefrontlib/src/cms-components/myaccount/order/cancellations-returns/cancel-or-return.service.ts]: prevent "return" button double clicked.

File: projects/core/src/user/facade/order-return-request.service.ts [hunk 4]
```diff
@@ -126,6 +127,20 @@ export class OrderReturnRequestService {
   }
 
   /**
+   * Get the order return request state
+   */
+  getReturnRequestState(): Observable<LoaderState<ReturnRequest>> {
+    return this.store.pipe(select(UsersSelectors.getOrderReturnRequestState));
+  }
```
[the remark this answers is not in the document]
Comment (WeizhengSap) [reply projects/core/src/user/facade/order-return-request.service.ts]: refactored

File: projects/storefrontlib/src/cms-components/myaccount/order/return-request-detail/return-request.service.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      tap(([returnCode, returnRequest, isLoading]) => {
+        if (
+          (returnRequest === undefined || returnRequest.rma !== returnCode) &&
+          !isLoading
+        ) {
+          this.returnRequestService.loadOrderReturnRequestDetail(returnCode);
+        }
+      }),
+      map(([_, returnRequest]) => returnRequest),
+      distinctUntilChanged()
```
[the remark this answers is not in the document]
Comment (WeizhengSap) [reply projects/storefrontlib/src/cms-components/myaccount/order/return-request-detail/return-request.service.ts]: This `distinctUntilChanged()` is not important. Without it, should work as well. The only purpose of it is, when `loading` flag changed, `ReturnRequest` data may not be changed. So, in this case, don't emit the same value.

File: projects/storefrontlib/src/cms-components/myaccount/order/return-request-detail/return-request-detail.module.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    RouterModule,
+    UrlModule,
+    I18nModule,
+    MediaModule,
+  ],
+  declarations: [...components],
+  exports: [...components],
+  entryComponents: [...components],
+})
+export class OrderReturnRequestDetailModule {}
```
[the remark this answers is not in the document]
Comment (WeizhengSap) [reply projects/storefrontlib/src/cms-components/myaccount/order/return-request-detail/return-request-detail.module.ts]: ok. I will change it. The reason `order` is not in file name is, these files are in `order` folder. So, if the files' name is `order-xxxx` is redundant. But, "OrderReturnRequest" is more clear than "ReturnRequest".

**Pull request merged**